src/vs/workbench/contrib/notebook/browser/contrib/cellStatusBar/executionStatusBarItemController.ts

```diff
@@ -233,27 +239,38 @@ class TimerCellStatusBarItem extends Disposable {
 	}
 
 	private async _update() {
-		let item: INotebookCellStatusBarItem | undefined;
+		let timerItem: INotebookCellStatusBarItem | undefined;
+		let warningItem: INotebookCellStatusBarItem | undefined;
 		const runState = this._executionStateService.getCellExecution(this._cell.uri);
 		const state = runState?.state;
+		const startTime = this._cell.internalMetadata.runStartTime;
+		const adjustment = this._cell.internalMetadata.runStartTimeAdjustment ?? 0;
+		const endTime = this._cell.internalMetadata.runEndTime;
+
 		if (runState?.didPause) {
-			item = undefined;
+			timerItem = undefined;
 		} else if (state === NotebookCellExecutionState.Executing) {
-			const startTime = this._cell.internalMetadata.runStartTime;
-			const adjustment = this._cell.internalMetadata.runStartTimeAdjustment;
 			if (typeof startTime === 'number') {
-				item = this._getTimeItem(startTime, Date.now(), adjustment);
+				timerItem = this._getTimeItem(startTime, Date.now(), adjustment);
 				this._scheduler.schedule();
 			}
 		} else if (!state) {
-			const startTime = this._cell.internalMetadata.runStartTime;
-			const endTime = this._cell.internalMetadata.runEndTime;
 			if (typeof startTime === 'number' && typeof endTime === 'number') {
-				item = this._getTimeItem(startTime, endTime, undefined, true);
+				timerItem = this._getTimeItem(startTime, endTime, undefined, true);
+
+				const timerDuration = Date.now() - endTime + adjustment;
```

Comment:
I'm confused about the timing logic, I think this branch will run at any random time in the future when execution is done, but you are calling `Date.now()` and counting that as time that the renderer is running?

Reply:
Yea, the rendering time logic is incorrect. It's an attempt at diffing between how long the timer was running and the time that we show after execution and is more accurately "VS Code Processing time". I think next we need to add proper events for timing what we want to present to the user.